### ISSUE TITLE
Generated C++ parser does not allow whitespace to be a token

### ISSUE THREAD

**Issue by Arm-Mike** [BNFC/bnfc#431]
The following token definition generates bad C++ parser.

`token Foo '\'' ["01x "]* '\'' ;`

The lex file generated is `<YYINITIAL>'( |0|1|x)*'        yylval._string = strdup(yytext); return _SYMB_171;` but it needs to be `<YYINITIAL>'(\ |0|1|x)*'       yylval._string = strdup(yytext); return _SYMB_171;` for it to compile.

The make file reports the following error message

```
flex -Pasl_ -oLexer.C ASL.l
ASL.l:219: unrecognized rule
ASL.l:220: unrecognized rule
make: *** [Lexer.C] Error 1
```

**Comment by andreasabel** [comment]
I can confirm that is broken for the C-family backends but works with the other backends (Haskell, Java, Ocaml).

**Comment by andreasabel** [comment]
@Arm-Mike : I released the fix as 2.9.4.1 just now!

**Comment by andreasabel** [comment]
> _Now on his way to Jerusalem, Jesus traveled along the border between Samaria and Galilee. As he was going into a village, ten men who had leprosy met him. They stood at a distance and called out in a loud voice, “Jesus, Master, have pity on us!”_
>
> _When he saw them, he said, “Go, show yourselves to the priests.” And as they went, they were cleansed._
>
> _One of them, when he saw he was healed, came back, praising God in a loud voice.  He threw himself at Jesus’ feet and thanked him—and he was a Samaritan._
>
> _Jesus asked, “Were not all ten cleansed? Where are the other nine? Has no one returned to give praise to God except this foreigner?” Then he said to him, “Rise and go; your faith has made you well.”_

Quoted from [Luke 17,11-19](https://www.biblegateway.com/passage/?search=Luke%2017%3A11-19&version=NIV).
